test/remove.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+var S = require('..');
+
+var eq = require('./internal/eq');
+var jsc = require('jsverify');
+
+
+test('remove', function() {
+
+  eq(typeof S.remove, 'function');
+  eq(S.remove.length, 2);
+  eq(S.remove.toString(), 'remove :: String -> StrMap a -> StrMap a');
+
+  function testBasic(key, strMap, strMapResult) {
+    eq(S.remove(key, strMap), strMapResult);
+  }
+
+  function testDoubleRemove() {
+    jsc.assert(jsc.forall(jsc.string, jsc.dict(jsc.number), function(key, map) {
+      var remove = S.remove(key);
+      return S.equals(remove(remove(map)), remove(map));
+    }), {tests: 1000});
+  }
+
+  function testRemove(key, strMap, strMapResult) {
+    testBasic(key, strMap, strMapResult);
+    testDoubleRemove(key, strMap);
+  }
+
+  testRemove('a', {}, {});
+  testRemove('a', {a: 1}, {});
+  testRemove('a', {b: 2}, {b: 2});
+  testRemove('b', {a: 1, b: 2}, {a: 1});
+  testRemove('a', {b: 2, c: 3}, {b: 2, c: 3});
+  testRemove('b', {a: 1, b: 2, c: 3}, {a: 1, c: 3});
```

Comment:
As mentioned above, let's test the property just 1000 times. This will mean we can remove `testRemove` and make assertions here directly:

```javascript
eq(S.remove('a', {}), {});
...
eq(S.remove('b', {a: 1, b: 2, c: 3}), {a: 1, c: 3});
```

I think we could achieve 100% coverage with fewer test cases. These should suffice:

```javascript
eq(S.remove('x', {x: 1, y: 2}), {y: 2});
eq(S.remove('y', {x: 1, y: 2}), {x: 1});
eq(S.remove('z', {x: 1, y: 2}), {x: 1, y: 2});
```

Reply:
I've made those changes locally. Are the `insert` tests excessive as well?

Reply:
I would say so. I like to be able to look at each test case and see what *it alone* tests. I see only two distinct scenarios: either the key is already present or it is not. We really only need two unit tests (in addition to the property tests), but I don't mind including a few more if it makes a nice `a`, `b`, `c` or `1`, `2`, `3` pattern.

Reply:
Updated pull request with 3 `insert` tests, making somewhat of an `a`, `b` / `1`, `2`, `3` pattern.